### DISCUSSION TITLE
Riverpod Versus Inherited Model

### DISCUSSION THREAD

**Discussion by StructPoltergeist** [rrousselGit/riverpod#3046] [General]
The Inherited Model - https://api.flutter.dev/flutter/widgets/InheritedModel-class.html

Why should I use Riverpod?

**Comment by Shreemanarjun** [comment]
You can check why provider created in first place (provider vs inherited widget):- [https://riverpod.dev/docs/from_provider/motivation#disposing-of-state-is-difficult](https://riverpod.dev/docs/from_provider/motivation#disposing-of-state-is-difficult)

You can check what riverpod overcome both inheritedwidget and provider limitations : [link](https://riverpod.dev/docs/essentials/faq#why-do-we-need-to-extend-consumerwidget-instead-of-using-the-raw-statelesswidget)

**Comment by StructPoltergeist** [comment]
Thank you @Shreemanarjun
